### PR TITLE
Make annotations focusable (bug 1851489)

### PR DESCRIPTION
**Pull request by calixteman**
When the annotation has a popup then the popup can be toggled in using the Enter key and hidden in using the Escape key.

### CODE REVIEW

File: src/display/annotation_layer.js [hunk 1]
```diff
@@ -206,6 +206,7 @@ class AnnotationElement {
 
     const container = document.createElement("section");
     container.setAttribute("data-annotation-id", data.id);
+    container.setAttribute("tabindex", DEFAULT_TAB_INDEX);
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:209]: Looking at existing code, could we use `container.tabIndex = DEFAULT_TAB_INDEX;` instead?
Furthermore, it seems strange to set this unconditionally and then remove it for certain Annotation-types. Can't we simply do the following instead?
```suggestion
    if (!(this instance of WidgetAnnotationElement)) {
      container.tabIndex = DEFAULT_TAB_INDEX;
    }
```

File: src/display/annotation_layer.js [hunk 2]
```diff
@@ -990,6 +991,7 @@ class WidgetAnnotationElement extends AnnotationElement {
     if (this.data.alternativeText) {
       this.container.title = this.data.alternativeText;
     }
+    this.containter.removeAttribute("tabIndex");
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:994]: There's a pretty obvious typo here, that ought to throw; did you forget to test this patch locally :-P
Besides, assuming that the comment above is implemented this line should no longer be necessary.

File: src/display/annotation_layer.js [hunk 3]
```diff
@@ -2044,6 +2048,11 @@ class PopupElement {
       }
     }
 
+    // Attach the event listener to toggle the popup with the keyboard.
+    for (const element of elements) {
+      element.container.addEventListener("keydown", this.#boundKeyDown);
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:2053]: Is there any situation where the `element` might not be renderable, in which case `element.container` won't exist?
With that in mind, would it be safer to guard against that situation?
```suggestion
      element.container?.addEventListener("keydown", this.#boundKeyDown);
```

File: src/display/annotation_layer.js [hunk 4]
```diff
@@ -990,6 +993,7 @@ class WidgetAnnotationElement extends AnnotationElement {
     if (this.data.alternativeText) {
       this.container.title = this.data.alternativeText;
     }
+    this.containter.removeAttribute("tabInddsfsdfex");
```
Comment (Snuffleupagus) [inline src/display/annotation_layer.js:996]: `this.containter === undefined", hence you're attempting to call `undefined.removeAttribute(...)` which should throw :-)

Besides, this line no longer ought to be necessary with the change above

Comment (calixteman) [reply src/display/annotation_layer.js:996]: Arff... I just forgot to remove the line....

**Review by Snuffleupagus — APPROVED** [review]
r=me, with passing tests; thank you!

**Pull request merged**